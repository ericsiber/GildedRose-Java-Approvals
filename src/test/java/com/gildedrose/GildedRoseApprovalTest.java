package com.gildedrose;

import org.approvaltests.Approvals;
import org.approvaltests.reporters.DiffReporter;
import org.approvaltests.reporters.UseReporter;
import org.junit.jupiter.api.Test;

@UseReporter(DiffReporter.class)
public class GildedRoseApprovalTest {

	@Test
	public void other() {

        Item[] items = new Item[] {
            new Item("foo", 0, 0),
            new Item("bar", 0, 1),
            new Item("qix", 1, 0),
            new Item("flop", 1, 1),
            new Item("boom", 0, 2),
            new Item("qual", 0, 100),
        };
        GildedRose app = new GildedRose(items);
        app.updateQuality();

		Approvals.verifyAll("Items", items);
	}

    @Test
    public void conjured() {

        Item[] items = new Item[] {
            new Item("Conjured Mana Cake", 0, 0),
            new Item("Conjured Mana Cake", 0, 4),
            new Item("Conjured Mana Cake", 0, 2),
            new Item("Conjured Mana Cake", 1, 4),
            new Item("Conjured Mana Cake", 0, 3),
        };
        GildedRose app = new GildedRose(items);
        app.updateQuality();

        Approvals.verifyAll("Items", items);
    }

    @Test
    public void AgedBrie() {

        Item[] items = new Item[] {
            new Item("Aged Brie", 2, 0),
            new Item("Aged Brie", 1, 1),
            new Item("Aged Brie", 0, 1),
            new Item("Aged Brie", 0, 49)
        };
        GildedRose app = new GildedRose(items);
        app.updateQuality();

        Approvals.verifyAll("Items", items);
    }

    @Test
    public void Sulfuras() {

        Item[] items = new Item[] {
            new Item("Sulfuras, Hand of Ragnaros", 0, 2),
            new Item("Sulfuras, Hand of Ragnaros", 1, 50),
            new Item("Sulfuras, Hand of Ragnaros", 1, 1),
            new Item("Sulfuras, Hand of Ragnaros", -1, 50),
            new Item("Sulfuras, Hand of Ragnaros", -1, 49),
        };
        GildedRose app = new GildedRose(items);
        app.updateQuality();

        Approvals.verifyAll("Items", items);

    }

    @Test
    public void backstagePass() {

        Item[] items = new Item[] {
            new Item("Backstage passes to a TAFKAL80ETC concert", 0, 2),
            new Item("Backstage passes to a TAFKAL80ETC concert", 1, 50),
            new Item("Backstage passes to a TAFKAL80ETC concert", 1, 1),
            new Item("Backstage passes to a TAFKAL80ETC concert", 10, 1),
            new Item("Backstage passes to a TAFKAL80ETC concert", 11, 1),
            new Item("Backstage passes to a TAFKAL80ETC concert", 5, 1),
            new Item("Backstage passes to a TAFKAL80ETC concert", 6, 1),
            new Item("Backstage passes to a TAFKAL80ETC concert", -1, 50),
        };
        GildedRose app = new GildedRose(items);
        app.updateQuality();

        Approvals.verifyAll("Items", items);

    }
}
