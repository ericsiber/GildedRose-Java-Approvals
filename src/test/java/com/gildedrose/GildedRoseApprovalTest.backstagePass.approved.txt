Items[0] = Backstage passes to a TAFKAL80ETC concert, -1, 0
Items[1] = Backstage passes to a TAFKAL80ETC concert, 0, 50
Items[2] = Backstage passes to a TAFKAL80ETC concert, 0, 4
Items[3] = Backstage passes to a TAFKAL80ETC concert, 9, 3
Items[4] = Backstage passes to a TAFKAL80ETC concert, 10, 2
Items[5] = Backstage passes to a TAFKAL80ETC concert, 4, 4
Items[6] = Backstage passes to a TAFKAL80ETC concert, 5, 3
Items[7] = Backstage passes to a TAFKAL80ETC concert, -2, 0
